framework/src/base/FEProblem.C

```diff
@@ -704,16 +709,47 @@ FEProblem::getMaxScalarOrder() const
 }
 
 void
+FEProblem::checkNonlocalCoupling()
```

Comment:
I agree - but it's too expensive to do all the time.  Going to have to find another way to do it.